Realm/ObjectServerTests/SwiftMongoClientTests.swift

```diff
@@ -0,0 +1,1222 @@
+        XCTAssertNotNil(deletedDocument)
+        XCTAssertEqual(deletedDocument?["name"]??.stringValue, "tomas") // shouldReturnNewDocument is false that is why returns old document
+
+        // Document already deleted, should return nil
+        let options2 = FindOneAndModifyOptions(["name": 1], ["_id": 1])
+        let deletedDocument3 = try await collection.findOneAndDelete(filter: document, options: options2)
+        XCTAssertNil(deletedDocument3)
+
+        let resultObjectIds = try await collection.find(filter: [:])
+        XCTAssertEqual(resultObjectIds.count, 3)
+    }
+
+    func testMongoCollectionUpdateOneAsyncAwait() async throws {
+        let collection = try await setupMongoCollection()
+
+        let document: Document = ["name": "tomas", "breed": "jack rusell"]
+        let document1: Document = ["name": "lucas", "breed": "german shepard"]
+        let document2: Document = ["name": "fito", "breed": "goberian"]
+        let document3: Document = ["name": "fosca", "breed": "labradoodle"]
+        let document4: Document = ["name": "balo", "breed": "pug"]
+        let document5: Document = ["name": "luna", "breed": "weimaraner"]
+
+        let objectIds = try await collection.insertMany([document, document1, document2, document3])
+        XCTAssertNotNil(objectIds)
+        XCTAssertEqual(objectIds.count, 4)
+
+        let updatedResult = try await collection.updateOneDocument(filter: document, update: document2)
+        XCTAssertNil(updatedResult.objectId) // No upsert
+        XCTAssertEqual(updatedResult.matchedCount, 1)
+        XCTAssertEqual(updatedResult.modifiedCount, 1)
+
+        let updatedResult1 = try await collection.updateOneDocument(filter: document1,
+                                                                    update: document3,
+                                                                    upsert: true)
+        XCTAssertNil(updatedResult1.objectId) // No upsert, already exist
+        XCTAssertEqual(updatedResult1.matchedCount, 1)
+        XCTAssertEqual(updatedResult1.modifiedCount, 1)
+
+        let updatedResult2 = try await collection.updateOneDocument(filter: document4,
+                                                                    update: document2,
+                                                                    upsert: true)
+        XCTAssertNotNil(updatedResult2.objectId)
+        XCTAssertEqual(updatedResult2.matchedCount, 0)
+        XCTAssertEqual(updatedResult2.modifiedCount, 0)
+
+        let updatedResult3 = try await collection.updateOneDocument(filter: document5,
+                                                                    update: document2)
+        XCTAssertNil(updatedResult3.objectId) // No upsert
+        XCTAssertEqual(updatedResult3.matchedCount, 0)
+        XCTAssertEqual(updatedResult3.modifiedCount, 0)
+    }
+
+    func testMongoCollectionUpdateManyAsyncAwait() async throws {
```

Comment:
I think this test is doing too much? Seems like the content can be cut in half and handled with one or two updates instead of four.